src/pages/MyPage/MyPage.js

```diff
@@ -22,16 +23,19 @@ export default class MyPage extends Component {
       .then(res => {
         if (res.status === 200) {
           return res.json();
-        }
+        } else alert('로그인해주세요');
       })
       .then(userRatedData => {
         this.setState({
-          userRatedData: userRatedData['result'],
+          userData: userRatedData.result,
         });
       });
   }
 
   render() {
+    const userName = this.state.userData[0]?.userName;
+    const moviesCount = this.state.userData[0]?.moviesCount;
+    const wantedCount = this.state.userData[0]?.wantedCount;
```

Comment:
옵셔널 체이닝도 활용해주셨군요??👍
다만, 알고 써야 합니다. 어떤 문법인지 코멘트로 설명 달아주세요
```suggestion
    const {userName, moviesCount, wantedCount} = this.state.userData[0]
```
으로 구조분해 해서 사용할 수 있어보이네요!
만약 초기값이 없어서 에러가뜬다면 || 연산자 잘 활용해보시면 해결할 수 있을겁니다

Reply:
옵셔널 체이닝은 앞 평가 대상이 존재하지 않으면 undefined를 반환하고 평가를 멈춥니다. 이렇게 되면 앞 평가 대상이 반드시 필요한 상황에서 존재하지 않는 에러를 캐치하기 어려워집니다. 그래서 앞 평가 대상이 꼭 필요한 경우에선 사용을 지양해야 합니다 ~ 우왕 👍🏻

Reply:
?. 좌측이 undefined 또는 null 상태라면 평가를 멈추고 undefined를 반환합니다. 그래서 라이프사이클에 의해 fetch 받는 데이터 특히 중첩객체 등에 접근할 때 모던 자바스크립트에 의하면 에러없이 안전하게~ 접근할 수 있습니다. 구조분해시에 저렇게 ||와 함께 하면 에러에 대응할 수 있다는 것도 처음 알았는데 너무 신기하고 좋은거 같아요 구조분해 아직 안 익숙해서 자유자재로 활용하기까진 시간이 걸릴거 같지만 😂 감사해요 연욱님~👍🏻 연욱님 코멘트를 보고 함께 고민해준 동기에게도 전달했는데 너무 좋아했습니다 크~